integrationtests/agent/bundle_deployment_status_test.go

```diff
@@ -0,0 +1,209 @@
+package agent
+
+import (
+	"time"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/wrangler/pkg/genericcondition"
+
+	"github.com/google/go-cmp/cmp"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	bundle           = "bundle"
+	timeout          = 5 * time.Second
```

Comment:
If we timeout is always the same, I'd prefer `SetDefaultEventuallyTimeout` in `suite_test.go`, over the more verbose `WithTimeout` (https://github.com/rancher/fleet/blob/master/e2e/single-cluster/suite_test.go#L22)

Reply:
https://github.com/rancher/fleet/pull/1183/commits/d1a0b45bf10cba03ffcf2da879e262e89b7330d6